crates/pipeline_manager/src/api/pipeline.rs

```diff
@@ -7,604 +17,481 @@ use actix_web::{
-        Ok(revision) => Some(revision),
-        Err(e) => {
-            if matches!(e, DBError::NoRevisionAvailable { .. }) {
-                None
-            } else {
-                Err(e)?
-            }
-        }
-    };
-
-    Ok(HttpResponse::Ok()
-        .insert_header(CacheControl(vec![CacheDirective::NoCache]))
-        .json(&descr))
-}
-
-/// Retrieve pipeline metrics and performance counters.
-#[utoipa::path(
-    responses(
-        // TODO: Implement `ToSchema` for `ControllerStatus`, which is the
-        // actual type returned by this endpoint.
-        (status = OK, description = "Pipeline metrics retrieved successfully.", body = Object),
-        (status = BAD_REQUEST
-            , description = "Specified pipeline id is not a valid uuid."
-            , body = ErrorResponse
-            , example = json!(examples::invalid_uuid_param())),
-        (status = NOT_FOUND
-            , description = "Specified pipeline id does not exist."
-            , body = ErrorResponse
-            , example = json!(examples::unknown_pipeline())),
-    ),
-    params(
-        ("pipeline_name" = String, Path, description = "Unique pipeline name"),
+    request_body(
+        content = PatchPipeline, example = json!(examples::patch_pipeline())
     ),
-    context_path = "/v0",
-    security(("JSON web token (JWT) or API key" = [])),
-    tag = "Pipelines"
-)]
-#[get("/pipelines/{pipeline_name}/stats")]
-pub(crate) async fn pipeline_stats(
-    state: WebData<ServerState>,
-    tenant_id: ReqData<TenantId>,
-    req: HttpRequest,
-) -> Result<HttpResponse, ManagerError> {
-    let pipeline_name = parse_string_param(&req, "pipeline_name")?;
-    state
-        .runner
-        .forward_to_pipeline(*tenant_id, &pipeline_name, Method::GET, "stats")
-        .await
-}
-
-/// Fetch a pipeline by ID.
-#[utoipa::path(
     responses(
-        (status = OK, description = "Pipeline descriptor retrieved successfully.",content(
-            ("application/json" = Pipeline, example = json!(examples::pipeline_config())),
-        )),
+        (status = OK
+            , description = "Pipeline successfully updated"
+            , body = ExtendedPipelineDescr
+            , example = json!(examples::extended_pipeline_1())),
         (status = NOT_FOUND
-            , description = "Specified pipeline ID does not exist."
+            , description = "Pipeline with that name does not exist"
             , body = ErrorResponse
-            , example = json!(examples::unknown_pipeline())),
-    ),
-    params(
-        ("pipeline_name" = String, Path, description = "Unique pipeline name"),
+            , example = json!(examples::error_unknown_pipeline())),
+        (status = CONFLICT
+            , description = "Cannot rename pipeline as the name already exists"
+            , body = ErrorResponse
+            , example = json!(examples::error_duplicate_name())),
+        (status = BAD_REQUEST
+            , description = "Pipeline is not shutdown"
```

Comment:
```suggestion
            , description = "Pipeline needs to be shutdown to be modified"
```

Reply:
Done, updated in three places (full update, partial update, and delete).